src/components/modals/SwapCoins/InputModal/index.tsx

```diff
@@ -0,0 +1,82 @@
+import Button from "components/inputs/Button";
+import InputAmount from "components/inputs/InputAmount";
+import ModalContainer from "components/modals/ModalContainer";
+import React, { useCallback, useState } from "react";
+import { InputModalProps, Swap } from "../types";
+import { ButtonContainer, IconContainer } from "./styles";
+import SwapIcon from "assets/icons/swap-icon.svg";
+import { dark } from "theme";
+import { Gap } from "utils/gap";
+
+const InputModal = ({
+  avaliableCoins,
+  continueButtonOnClick = () => {},
+  onClickClose = () => {},
+  getSwapRate = () => {},
+}: InputModalProps) => {
+  const [swapFrom, setSwapFrom] = useState<Swap>();
+  const [swapTo, setSwapTo] = useState<Swap>();
+  const [intervalId, setIntervalId] = useState<any>();
+  const rate = getSwapRate({ swapFrom: swapFrom, swapTo: swapTo });
```

Comment:
Use useMemo

Reply:
Ah this is when it should be useMemo. Gotcha.